zerver/management/commands/makemessages.py

```diff
@@ -247,7 +246,6 @@ def get_new_strings(
         """
         new_strings = {}  # Dict[str, str]
         for k in translation_strings:
-            k = k.replace("\\n", "\n")
```

Comment:
e710110a9ea94716b4eeca396e6f8fb4c62c02d3 appears to be the background for this.  I imagine it's likely that FormatJS handles this issue in a better way that doesn't include the leading/training whitespace in our handlebars translation blocks in what we ask translators to deal with, but perhaps worth checking.

As a sidenote, have you attempt to do any sort of audit with having FormatJS output a `translations.json` file and diffing that against the old English version from before this PR?  I think that'd be a good way to verify this PR only improves things.

Reply:
`translations.json` diff:

<details>

```diff
@@ -5,6 +5,7 @@
   "(no topic)": "(no topic)",
   "(unavailable)": "(unavailable)",
   "(you)": "(you)",
+  "({message_retention_days} days)": "({message_retention_days} days)",
   "/dark (Toggle night mode)": "/dark (Toggle night mode)",
   "/day (Toggle day mode)": "/day (Toggle day mode)",
   "/fixed-width (Toggle fixed width mode)": "/fixed-width (Toggle fixed width mode)",
@@ -23,11 +24,13 @@
   "2 minutes": "2 minutes",
   "24-hour clock (17:00)": "24-hour clock (17:00)",
   "3 days": "3 days",
-  "<a href=\"/help/export-your-organization\" target=\"_blank\" rel=\"noopener noreferrer\">Click here</a> to learn about exporting private streams and messages.": "<a href=\"/help/export-your-organization\" target=\"_blank\" rel=\"noopener noreferrer\">Click here</a> to learn about exporting private streams and messages.",
-  "<a href=\"/upgrade\" target=\"_blank\" rel=\"noopener noreferrer\">Upgrade</a> for more space.": "<a href=\"/upgrade\" target=\"_blank\" rel=\"noopener noreferrer\">Upgrade</a> for more space.",
-  "<p>Stream will be announced in <b>#__notifications_stream__</b>.</p>": "<p>Stream will be announced in <b>#__notifications_stream__</b>.</p>",
-  "<strong>__name__</strong> is not subscribed to this stream. They will not be notified if you mention them.": "<strong>__name__</strong> is not subscribed to this stream. They will not be notified if you mention them.",
-  "<strong>__name__</strong> is not subscribed to this stream. They will not be notified unless you subscribe them.": "<strong>__name__</strong> is not subscribed to this stream. They will not be notified unless you subscribe them.",
+  "<b>Total messages</b>: {total_messages}": "<b>Total messages</b>: {total_messages}",
+  "<p>Stream will be announced in <b>#{notifications_stream}</b>.</p>": "<p>Stream will be announced in <b>#{notifications_stream}</b>.</p>",
+  "<p>The stream <b>{stream_name}</b> does not exist.</p><p>Manage your subscriptions <z-link>on your Streams page</z-link>.</p>": "<p>The stream <b>{stream_name}</b> does not exist.</p><p>Manage your subscriptions <z-link>on your Streams page</z-link>.</p>",
+  "<strong>{name}</strong> is not subscribed to this stream. They will not be notified if you mention them.": "<strong>{name}</strong> is not subscribed to this stream. They will not be notified if you mention them.",
+  "<strong>{name}</strong> is not subscribed to this stream. They will not be notified unless you subscribe them.": "<strong>{name}</strong> is not subscribed to this stream. They will not be notified unless you subscribe them.",
+  "<z-link>Click here</z-link> to learn about exporting private streams and messages.": "<z-link>Click here</z-link> to learn about exporting private streams and messages.",
+  "<z-link>Upgrade</z-link> for more space.": "<z-link>Upgrade</z-link> for more space.",
   "A Topic Move already in progress.": "A Topic Move already in progress.",
   "A stream needs to have a name": "A stream needs to have a name",
   "A stream with this name already exists": "A stream with this name already exists",
@@ -53,7 +56,7 @@
   "Add emoji": "Add emoji",
   "Add emoji reaction": "Add emoji reaction",
   "Add emoji reaction (:)": "Add emoji reaction (:)",
-  "Add extra emoji for members of the __realm_name__ organization.": "Add extra emoji for members of the __realm_name__ organization.",
+  "Add extra emoji for members of the {realm_name} organization.": "Add extra emoji for members of the {realm_name} organization.",
   "Add linkifier": "Add linkifier",
   "Add member...": "Add member...",
   "Add new default stream": "Add new default stream",
@@ -74,6 +77,7 @@
   "Admins, members and guests": "Admins, members and guests",
   "Admins, members, and guests": "Admins, members, and guests",
   "Alert word": "Alert word",
+  "Alert word \"{word}\" added successfully!": "Alert word \"{word}\" added successfully!",
   "Alert word already exists!": "Alert word already exists!",
   "Alert word can't be empty!": "Alert word can't be empty!",
   "Alert word removed successfully!": "Alert word removed successfully!",
@@ -86,36 +90,38 @@
   "All streams": "All streams",
   "All time": "All time",
   "All unreads": "All unreads",
+  "Allow message content in missed message emails": "Allow message content in missed message emails",
   "Allow message deleting": "Allow message deleting",
   "Allow message editing": "Allow message editing",
   "Allow subdomains": "Allow subdomains",
   "Allowed domains": "Allowed domains",
-  "Allowed domains: __domains__": "Allowed domains: __domains__",
-  "Already subscribed to __stream__": "Already subscribed to __stream__",
+  "Allowed domains: {domains}": "Allowed domains: {domains}",
+  "Already subscribed to {stream}": "Already subscribed to {stream}",
   "Already subscribed users:": "Already subscribed users:",
   "Always": "Always",
-  "An API key can be used to programmatically access a Zulip account. Anyone with access to your API key has the ability to read your messages, send messages on your behalf, and otherwise impersonate you on Zulip, so you should guard your API key as carefully as you guard your password. <br> We recommend creating bots and using the bots' accounts and API keys to access the Zulip API, unless the task requires access to your account.": "An API key can be used to programmatically access a Zulip account. Anyone with access to your API key has the ability to read your messages, send messages on your behalf, and otherwise impersonate you on Zulip, so you should guard your API key as carefully as you guard your password. <br> We recommend creating bots and using the bots' accounts and API keys to access the Zulip API, unless the task requires access to your account.",
+  "An API key can be used to programmatically access a Zulip account. Anyone with access to your API key has the ability to read your messages, send messages on your behalf, and otherwise impersonate you on Zulip, so you should guard your API key as carefully as you guard your password. <br /> We recommend creating bots and using the bots' accounts and API keys to access the Zulip API, unless the task requires access to your account.": "An API key can be used to programmatically access a Zulip account. Anyone with access to your API key has the ability to read your messages, send messages on your behalf, and otherwise impersonate you on Zulip, so you should guard your API key as carefully as you guard your password. <br /> We recommend creating bots and using the bots' accounts and API keys to access the Zulip API, unless the task requires access to your account.",
   "An hour ago": "An hour ago",
   "An unknown error occurred.": "An unknown error occurred.",
   "Announce stream": "Announce stream",
   "Any member of this organization can add custom emoji.": "Any member of this organization can add custom emoji.",
   "Any organization administrator can conduct an export.": "Any organization administrator can conduct an export.",
   "Any time": "Any time",
+  "Anyone can join; anyone can view complete message history without joining": "Anyone can join; anyone can view complete message history without joining",
   "Anyone in this organization can add bots": "Anyone in this organization can add bots",
   "April": "April",
   "Archive stream": "Archive stream",
   "Archiving this stream will immediately unsubscribe everyone, and this action cannot be undone.": "Archiving this stream will immediately unsubscribe everyone, and this action cannot be undone.",
   "Are invitations required for joining the organization?": "Are invitations required for joining the organization?",
-  "Are you sure you want to create stream '__stream_name__' and subscribe __count__ users to it?": "Are you sure you want to create stream '__stream_name__' and subscribe __count__ users to it?",
-  "Are you sure you want to delete <b>__group_name__</b>?": "Are you sure you want to delete <b>__group_name__</b>?",
+  "Are you sure you want to create stream ''''{stream_name}'''' and subscribe {count} users to it?": "Are you sure you want to create stream ''''{stream_name}'''' and subscribe {count} users to it?",
+  "Are you sure you want to delete <b>{group_name}</b>?": "Are you sure you want to delete <b>{group_name}</b>?",
   "Are you sure you want to delete your profile picture?": "Are you sure you want to delete your profile picture?",
   "Are you sure you want to do this?": "Are you sure you want to do this?",
-  "Are you sure you want to mention all <strong>__count__</strong> people in this stream? <br> This will send email and mobile push notifications to most of those <strong>__count__</strong> users. <br> If you don't want to do that, please edit your message to remove the <strong>@__mention__</strong> mention.": "Are you sure you want to mention all <strong>__count__</strong> people in this stream? <br> This will send email and mobile push notifications to most of those <strong>__count__</strong> users. <br> If you don't want to do that, please edit your message to remove the <strong>@__mention__</strong> mention.",
-  "Are you sure you want to permanently delete <b>__topic_name__</b>?": "Are you sure you want to permanently delete <b>__topic_name__</b>?",
-  "Are you sure you want to resend the invitation to <strong><span class=\"email\"></span></strong>?": "Are you sure you want to resend the invitation to <strong><span class=\"email\"></span></strong>?",
-  "Are you sure you want to revoke the invitation to <strong>__email__</strong>?": "Are you sure you want to revoke the invitation to <strong>__email__</strong>?",
-  "Are you sure you want to revoke this invitation link created by <strong>__referred_by__</strong>?": "Are you sure you want to revoke this invitation link created by <strong>__referred_by__</strong>?",
-  "Are you sure you want to unstar all messages in topic <b>__topic__</b>?  This action cannot be undone.": "Are you sure you want to unstar all messages in topic <b>__topic__</b>?  This action cannot be undone.",
+  "Are you sure you want to mention all <strong>{count}</strong> people in this stream? <br> This will send email and mobile push notifications to most of those <strong>{count}</strong> users. <br> If you don't want to do that, please edit your message to remove the <strong>@{mention}</strong> mention.": "Are you sure you want to mention all <strong>{count}</strong> people in this stream? <br> This will send email and mobile push notifications to most of those <strong>{count}</strong> users. <br> If you don't want to do that, please edit your message to remove the <strong>@{mention}</strong> mention.",
+  "Are you sure you want to permanently delete <b>{topic_name}</b>?": "Are you sure you want to permanently delete <b>{topic_name}</b>?",
+  "Are you sure you want to resend the invitation to <z-email></z-email>?": "Are you sure you want to resend the invitation to <z-email></z-email>?",
+  "Are you sure you want to revoke the invitation to <strong>{email}</strong>?": "Are you sure you want to revoke the invitation to <strong>{email}</strong>?",
+  "Are you sure you want to revoke this invitation link created by <strong>{referred_by}</strong>?": "Are you sure you want to revoke this invitation link created by <strong>{referred_by}</strong>?",
+  "Are you sure you want to unstar all messages in topic <b>{topic}</b>?  This action cannot be undone.": "Are you sure you want to unstar all messages in topic <b>{topic}</b>?  This action cannot be undone.",
   "Attach files": "Attach files",
   "Attachment deleted": "Attachment deleted",
   "Audible desktop notifications": "Audible desktop notifications",
@@ -130,7 +136,7 @@
   "Bot email (a-z, 0-9, and dashes only)": "Bot email (a-z, 0-9, and dashes only)",
   "Bot type": "Bot type",
   "Bots": "Bots",
-  "By deactivating <strong><span class=\"user_name\"></span></strong> &lt;<span class=\"email\"></span>&gt;, they will be logged out immediately.": "By deactivating <strong><span class=\"user_name\"></span></strong> &lt;<span class=\"email\"></span>&gt;, they will be logged out immediately.",
+  "By deactivating <z-user></z-user>, they will be logged out immediately.": "By deactivating <z-user></z-user>, they will be logged out immediately.",
   "By deactivating your account, you will be logged out immediately.": "By deactivating your account, you will be logged out immediately.",
   "Cancel": "Cancel",
   "Change": "Change",
@@ -145,6 +151,7 @@
   "Change stream permissions for #": "Change stream permissions for #",
   "Change user info and roles": "Change user info and roles",
   "Check all": "Check all",
+  "Check your email ({email}) to confirm the new address.": "Check your email ({email}) to confirm the new address.",
   "Choose avatar": "Choose avatar",
   "Clear emoji image": "Clear emoji image",
   "Clear profile picture": "Clear profile picture",
@@ -162,6 +169,7 @@
   "Configure the authentication methods for your organization.": "Configure the authentication methods for your organization.",
   "Configure the default streams new users are subscribed to when joining your organization.": "Configure the default streams new users are subscribed to when joining your organization.",
   "Confirm": "Confirm",
+  "Convert emoticons before sending (<code>:)</code> becomes \ud83d\ude03)": "Convert emoticons before sending (<code>:)</code> becomes \ud83d\ude03)",
   "Cookie Bot": "Cookie Bot",
   "Copied!": "Copied!",
   "Copy and close": "Copy and close",
@@ -197,6 +205,7 @@
   "Deactivated": "Deactivated",
   "Deactivated users": "Deactivated users",
   "December": "December",
+  "Default is {language}. Use 'text' to disable highlighting.": "Default is {language}. Use 'text' to disable highlighting.",
   "Default language": "Default language",
   "Default language for code blocks:": "Default language for code blocks:",
   "Default settings for new users joining this organization.": "Default settings for new users joining this organization.",
@@ -234,8 +243,8 @@
   "Download file": "Download file",
   "Download zuliprc": "Download zuliprc",
   "Drafts": "Drafts",
-  "Drafts (__draft_count__)": "Drafts (__draft_count__)",
-  "Drafts older than <strong>__draft_lifetime__</strong> days are automatically removed.": "Drafts older than <strong>__draft_lifetime__</strong> days are automatically removed.",
+  "Drafts ({draft_count})": "Drafts ({draft_count})",
+  "Drafts older than <strong>{draft_lifetime}</strong> days are automatically removed.": "Drafts older than <strong>{draft_lifetime}</strong> days are automatically removed.",
   "EDITED": "EDITED",
   "Edit": "Edit",
   "Edit bot": "Edit bot",
@@ -243,7 +252,7 @@
   "Edit topic": "Edit topic",
   "Edit user": "Edit user",
   "Edit your profile": "Edit your profile",
-  "Edited (__last_edit_timestr__)": "Edited (__last_edit_timestr__)",
+  "Edited ({last_edit_timestr})": "Edited ({last_edit_timestr})",
   "Email": "Email",
   "Email address": "Email address",
   "Email address changes are disabled in this organization.": "Email address changes are disabled in this organization.",
@@ -268,6 +277,8 @@
   "Error removing subscription": "Error removing subscription",
   "Error removing user from this stream.": "Error removing user from this stream.",
   "Error saving edit": "Error saving edit",
+  "Error: Cannot deactivate the only organization owner.": "Error: Cannot deactivate the only organization owner.",
+  "Error: Cannot deactivate the only user. You can deactivate the whole organization though in your <z-link>organization profile settings</z-link>.": "Error: Cannot deactivate the only user. You can deactivate the whole organization though in your <z-link>organization profile settings</z-link>.",
   "Estimated messages per week": "Estimated messages per week",
   "Everyone": "Everyone",
   "Expand message (-)": "Expand message (-)",
@@ -279,13 +290,14 @@
   "Failed": "Failed",
   "Failed to create video call.": "Failed to create video call.",
   "Failed to generate preview": "Failed to generate preview",
-  "Failed to upload %{file}": "Failed to upload %{file}",
+  "Failed to upload %'{file}'": "Failed to upload %'{file}'",
   "Failed!": "Failed!",
   "Failed: Emoji name is required.": "Failed: Emoji name is required.",
   "February": "February",
   "Field choices": "Field choices",
   "File": "File",
   "File and image uploads have been disabled for this organization.": "File and image uploads have been disabled for this organization.",
+  "File size must be at most {max_file_size} MiB.": "File size must be at most {max_file_size} MiB.",
   "File type is not supported.": "File type is not supported.",
   "Filter": "Filter",
   "Filter bots": "Filter bots",
@@ -299,7 +311,7 @@
   "Filter streams": "Filter streams",
   "Filter topics (t)": "Filter topics (t)",
   "Filter users": "Filter users",
-  "First time? Read our <a href=\"/help/getting-your-organization-started-with-zulip#create-streams\" target=\"_blank\" rel=\"noopener noreferrer\">guidelines</a> for creating and naming streams.": "First time? Read our <a href=\"/help/getting-your-organization-started-with-zulip#create-streams\" target=\"_blank\" rel=\"noopener noreferrer\">guidelines</a> for creating and naming streams.",
+  "First time? Read our <z-link>guidelines</z-link> for creating and naming streams.": "First time? Read our <z-link>guidelines</z-link> for creating and naming streams.",
   "Fixed width": "Fixed width",
   "Fixed width mode": "Fixed width mode",
   "Fluid width": "Fluid width",
@@ -327,12 +339,16 @@
   "Idle": "Idle",
   "Image": "Image",
   "Inactive bots": "Inactive bots",
+  "Include content of private messages in desktop notifications": "Include content of private messages in desktop notifications",
+  "Include message content in missed message emails": "Include message content in missed message emails",
   "Include muted": "Include muted",
+  "Include organization name in subject of missed message emails": "Include organization name in subject of missed message emails",
   "Incoming webhooks can only send messages.": "Incoming webhooks can only send messages.",
   "Interface": "Interface",
   "Invalid URL": "Invalid URL",
+  "Invalid slash command. Check if you are missing a space after the command.": "Invalid slash command. Check if you are missing a space after the command.",
   "Invalid stream id": "Invalid stream id",
-  "Invalid time format: __timestamp__": "Invalid time format: __timestamp__",
+  "Invalid time format: {timestamp}": "Invalid time format: {timestamp}",
   "Invite": "Invite",
   "Invite link": "Invite link",
   "Invite more users": "Invite more users",
@@ -357,11 +373,11 @@
   "Last 30 days": "Last 30 days",
   "Last 6 months": "Last 6 months",
   "Last active": "Last active",
-  "Last active: __- last_seen__": "Last active: __- last_seen__",
+  "Last active: {last_seen}": "Last active: {last_seen}",
   "Last modified": "Last modified",
   "Link:": "Link:",
   "Local time": "Local time",
-  "Looking for our <a href=\"/integrations\" target=\"_blank\" rel=\"noopener noreferrer\">Integrations</a> or <a href=\"/api\" rel=\"noopener noreferrer\" target=\"_blank\">API</a> documentation?": "Looking for our <a href=\"/integrations\" target=\"_blank\" rel=\"noopener noreferrer\">Integrations</a> or <a href=\"/api\" rel=\"noopener noreferrer\" target=\"_blank\">API</a> documentation?",
+  "Looking for our <z-integrations>integrations</z-integrations> or <z-api>API</z-api> documentation?": "Looking for our <z-integrations>integrations</z-integrations> or <z-api>API</z-api> documentation?",
   "Manage organization": "Manage organization",
   "Manage user groups": "Manage user groups",
   "March": "March",
@@ -375,8 +391,8 @@
   "Mention a timezone-aware time": "Mention a timezone-aware time",
   "Mentioned in": "Mentioned in",
   "Mentions": "Mentions",
-  "Message #__- stream_name__": "Message #__- stream_name__",
-  "Message __- recipient_names__": "Message __- recipient_names__",
+  "Message #{stream_name}": "Message #{stream_name}",
+  "Message #{stream_name} > {topic_name}": "Message #{stream_name} > {topic_name}",
   "Message actions": "Message actions",
   "Message editing": "Message editing",
   "Message formatting": "Message formatting",
@@ -384,26 +400,32 @@
   "Message retention for stream": "Message retention for stream",
   "Message retention period": "Message retention period",
   "Message sent when you were not subscribed": "Message sent when you were not subscribed",
+  "Message {recipient_names}": "Message {recipient_names}",
+  "Message {recipient_name} ({recipient_status})": "Message {recipient_name} ({recipient_status})",
+  "Messages in this stream will be automatically deleted after {retention_days} days.": "Messages in this stream will be automatically deleted after {retention_days} days.",
   "Messages in this stream will be retained forever.": "Messages in this stream will be retained forever.",
   "Method": "Method",
   "Mobile": "Mobile",
   "Mobile notifications": "Mobile notifications",
   "Mobile push notifications are not configured on this server.": "Mobile push notifications are not configured on this server.",
   "Monday": "Monday",
-  "More details are available <a href=\"/help/add-a-custom-linkifier\" target=\"_blank\" rel=\"noopener noreferrer\">in the Help Center article</a>.": "More details are available <a href=\"/help/add-a-custom-linkifier\" target=\"_blank\" rel=\"noopener noreferrer\">in the Help Center article</a>.",
+  "More details are available <z-link>in the Help Center article</z-link>.": "More details are available <z-link>in the Help Center article</z-link>.",
   "More than 2 weeks ago": "More than 2 weeks ago",
-  "Move all messages in <strong>__topic_name__</strong>": "Move all messages in <strong>__topic_name__</strong>",
+  "Move all messages in <strong>{topic_name}</strong>": "Move all messages in <strong>{topic_name}</strong>",
   "Move topic": "Move topic",
+  "Must be invited by a member; new members can only see messages sent after they join; hidden from non-administrator users": "Must be invited by a member; new members can only see messages sent after they join; hidden from non-administrator users",
+  "Must be invited by a member; new members can view complete message history; hidden from non-administrator users": "Must be invited by a member; new members can view complete message history; hidden from non-administrator users",
   "Mute stream": "Mute stream",
-  "Mute the topic <b>__topic__</b>": "Mute the topic <b>__topic__</b>",
+  "Mute the topic <b>{topic}</b>": "Mute the topic <b>{topic}</b>",
   "Mute topic": "Mute topic",
   "Muted streams don't show up in \"All messages\" or generate notifications unless you are mentioned.": "Muted streams don't show up in \"All messages\" or generate notifications unless you are mentioned.",
   "N": "N",
   "Name": "Name",
   "Name changes are disabled in this organization. Contact an administrator to change your name.": "Name changes are disabled in this organization. Contact an administrator to change your name.",
-  "Narrow to stream &quot;__display_recipient__&quot;": "Narrow to stream &quot;__display_recipient__&quot;",
-  "Narrow to stream &quot;__display_recipient__&quot;, topic &quot;__topic__&quot;": "Narrow to stream &quot;__display_recipient__&quot;, topic &quot;__topic__&quot;",
-  "Narrow to your private messages with __display_reply_to__": "Narrow to your private messages with __display_reply_to__",
+  "Narrow to stream &quot;{display_recipient}&quot;": "Narrow to stream &quot;{display_recipient}&quot;",
+  "Narrow to stream &quot;{display_recipient}&quot;, topic &quot;{topic}&quot;": "Narrow to stream &quot;{display_recipient}&quot;, topic &quot;{topic}&quot;",
+  "Narrow to your private messages with {display_reply_to}": "Narrow to your private messages with {display_reply_to}",
+  "Narrow to {message_recipient}": "Narrow to {message_recipient}",
   "Never": "Never",
   "Never had one? Forgotten it?": "Never had one? Forgotten it?",
   "New": "New",
@@ -411,6 +433,7 @@
   "New choice": "New choice",
   "New email": "New email",
   "New full name": "New full name",
+  "New members are not allowed to post to this stream.<br />Permission will be granted in {days} days.": "New members are not allowed to post to this stream.<br />Permission will be granted in {days} days.",
   "New members can only see messages sent after they join.": "New members can only see messages sent after they join.",
   "New members can view complete message history.": "New members can view complete message history.",
   "New option": "New option",
@@ -483,18 +506,19 @@
   "Organization profile": "Organization profile",
   "Organization profile picture": "Organization profile picture",
   "Organization settings": "Organization settings",
-  "Organization using __percent_used__% of __upload_quota__.": "Organization using __percent_used__% of __upload_quota__.",
+  "Organization using {percent_used}% of {upload_quota}.": "Organization using {percent_used}% of {upload_quota}.",
   "Other notification settings": "Other notification settings",
   "Other permissions": "Other permissions",
   "Other settings": "Other settings",
   "Outgoing webhook message format": "Outgoing webhook message format",
   "Owner": "Owner",
-  "Owner: __name__": "Owner: __name__",
+  "Owner: {name}": "Owner: {name}",
   "PMs, mentions, and alerts": "PMs, mentions, and alerts",
   "Participants": "Participants",
   "Participated": "Participated",
   "Password": "Password",
   "Password is too weak": "Password is too weak",
+  "Password should be at least {length} characters long": "Password should be at least {length} characters long",
   "Pattern": "Pattern",
   "People to add": "People to add",
   "Personal settings": "Personal settings",
@@ -541,7 +565,7 @@
   "Requesting user": "Requesting user",
   "Require topics in stream messages": "Require topics in stream messages",
   "Resend": "Resend",
-  "Resend invitation to <span class=\"email\"></span>": "Resend invitation to <span class=\"email\"></span>",
+  "Resend invitation to <z-email></z-email>": "Resend invitation to <z-email></z-email>",
   "Resend now": "Resend now",
   "Resending encountered an error. Please reload and try again.": "Resending encountered an error. Please reload and try again.",
   "Restore draft": "Restore draft",
@@ -552,7 +576,7 @@
   "Retry": "Retry",
   "Revoke": "Revoke",
   "Revoke invitation link": "Revoke invitation link",
-  "Revoke invitation to __email__": "Revoke invitation to __email__",
+  "Revoke invitation to {email}": "Revoke invitation to {email}",
   "Revoke now": "Revoke now",
   "Role": "Role",
   "SAVING": "SAVING",
@@ -561,6 +585,7 @@
   "Save changes": "Save changes",
   "Save failed": "Save failed",
   "Saved": "Saved",
+  "Saved. Please <z-link>reload</z-link> for the change to take effect.": "Saved. Please <z-link>reload</z-link> for the change to take effect.",
   "Saving": "Saving",
   "Scheduling...": "Scheduling...",
   "Search": "Search",
@@ -576,6 +601,7 @@
   "Send digest emails when I'm away": "Send digest emails when I'm away",
   "Send email notifications for new logins to my account": "Send email notifications for new logins to my account",
   "Send emails introducing Zulip to new users": "Send emails introducing Zulip to new users",
+  "Send mobile notifications even if I'm online (useful for testing)": "Send mobile notifications even if I'm online (useful for testing)",
   "Send notification to new topic": "Send notification to new topic",
   "Send notification to old topic": "Send notification to old topic",
   "Send private message": "Send private message",
@@ -609,6 +635,9 @@
   "Slack compatible": "Slack compatible",
   "Slack's outgoing webhooks": "Slack's outgoing webhooks",
   "Some common words were excluded from your search.": "Some common words were excluded from your search.",
+  "Sort by estimated weekly traffic": "Sort by estimated weekly traffic",
+  "Sort by name": "Sort by name",
+  "Sort by number of subscribers": "Sort by number of subscribers",
   "Spoiler": "Spoiler",
   "Star": "Star",
   "Starred messages": "Starred messages",
@@ -633,25 +662,26 @@
   "Sunday": "Sunday",
   "System bot": "System bot",
   "Task already exists": "Task already exists",
-  "The recipient __recipient__ is not valid": "The recipient __recipient__ is not valid",
-  "The recipients __recipients__ are not valid": "The recipients __recipients__ are not valid",
+  "The recipient {recipient} is not valid": "The recipient {recipient} is not valid",
+  "The recipients {recipients} are not valid": "The recipients {recipients} are not valid",
   "The stream description cannot contain newline characters.": "The stream description cannot contain newline characters.",
   "The stream description has been updated!": "The stream description has been updated!",
   "The stream has been renamed!": "The stream has been renamed!",
   "Their password will be cleared from our systems, and any bots they maintain will be disabled.": "Their password will be cleared from our systems, and any bots they maintain will be disabled.",
   "There are no current alert words.": "There are no current alert words.",
   "There are no messages to reply to.": "There are no messages to reply to.",
-  "These settings are explained in detail in the <a target=\"_blank\" rel=\"noopener noreferrer\" href=\"/help/stream-permissions\">help center</a>.": "These settings are explained in detail in the <a target=\"_blank\" rel=\"noopener noreferrer\" href=\"/help/stream-permissions\">help center</a>.",
+  "These settings are explained in detail in the <z-link>help center</z-link>.": "These settings are explained in detail in the <z-link>help center</z-link>.",
   "This action is permanent and cannot be undone. All users will permanently lose access to their Zulip accounts.": "This action is permanent and cannot be undone. All users will permanently lose access to their Zulip accounts.",
   "This file exceeds maximum allowed size of": "This file exceeds maximum allowed size of",
-  "This is a <i class=\"fa fa-globe\" aria-hidden=\"true\"></i> <b>web public stream</b>. Any member of the organization can join without an invitation and anyone on the internet can read the content published.": "This is a <i class=\"fa fa-globe\" aria-hidden=\"true\"></i> <b>web public stream</b>. Any member of the organization can join without an invitation and anyone on the internet can read the content published.",
-  "This is a <i class=\"hash\" aria-hidden=\"true\"></i> <b>public stream</b>. Any member of the organization can join without an invitation.": "This is a <i class=\"hash\" aria-hidden=\"true\"></i> <b>public stream</b>. Any member of the organization can join without an invitation.",
-  "This is a <span class=\"fa fa-lock\" aria-hidden=\"true\"></span> <b>private stream</b>. Only people who have been invited can access its content, but any member of the stream can invite others.": "This is a <span class=\"fa fa-lock\" aria-hidden=\"true\"></span> <b>private stream</b>. Only people who have been invited can access its content, but any member of the stream can invite others.",
+  "This is a <z-icon></z-icon> <b>private stream</b>. Only people who have been invited can access its content, but any member of the stream can invite others.": "This is a <z-icon></z-icon> <b>private stream</b>. Only people who have been invited can access its content, but any member of the stream can invite others.",
+  "This is a <z-icon></z-icon> <b>public stream</b>. Any member of the organization can join without an invitation.": "This is a <z-icon></z-icon> <b>public stream</b>. Any member of the organization can join without an invitation.",
+  "This is a <z-icon></z-icon> <b>web public stream</b>. Any member of the organization can join without an invitation and anyone on the internet can read the content published.": "This is a <z-icon></z-icon> <b>web public stream</b>. Any member of the organization can join without an invitation and anyone on the internet can read the content published.",
   "This is a private stream": "This is a private stream",
   "This is what a Zulip notification looks like.": "This is what a Zulip notification looks like.",
-  "This organization is configured to restrict editing of message content to __minutes_to_edit__ minutes after it is sent.": "This organization is configured to restrict editing of message content to __minutes_to_edit__ minutes after it is sent.",
+  "This organization is configured to restrict editing of message content to {minutes_to_edit} minutes after it is sent.": "This organization is configured to restrict editing of message content to {minutes_to_edit} minutes after it is sent.",
+  "This stream does not exist or is private.": "This stream does not exist or is private.",
   "This stream has been deactivated": "This stream has been deactivated",
-  "This stream is reserved for <strong>announcements</strong>. <br> Are you sure you want to message all <strong>__count__</strong> people in this stream?": "This stream is reserved for <strong>announcements</strong>. <br> Are you sure you want to message all <strong>__count__</strong> people in this stream?",
+  "This stream is reserved for <strong>announcements</strong>. <br> Are you sure you want to message all <strong>{count}</strong> people in this stream?": "This stream is reserved for <strong>announcements</strong>. <br> Are you sure you want to message all <strong>{count}</strong> people in this stream?",
   "Thursday": "Thursday",
   "Time": "Time",
   "Time format": "Time format",
@@ -659,6 +689,7 @@
   "Time zone": "Time zone",
   "Time's up!": "Time's up!",
   "Tip: You can also send \"/poll Some question\"": "Tip: You can also send \"/poll Some question\"",
+  "To add syntax highlighting to a multi-line code block, add the language's <b>first</b> <z-link>Pygments short name</z-link> after the first set of back-ticks. You can also make a code block by indenting each line with 4 spaces.": "To add syntax highlighting to a multi-line code block, add the language's <b>first</b> <z-link>Pygments short name</z-link> after the first set of back-ticks. You can also make a code block by indenting each line with 4 spaces.",
   "Today": "Today",
   "Toggle subscription": "Toggle subscription",
   "Tomorrow": "Tomorrow",
@@ -678,31 +709,34 @@
   "Unknown stream": "Unknown stream",
   "Unmute": "Unmute",
   "Unmute stream": "Unmute stream",
-  "Unmute the topic <b>__topic__</b>": "Unmute the topic <b>__topic__</b>",
+  "Unmute the topic <b>{topic}</b>": "Unmute the topic <b>{topic}</b>",
   "Unmute topic": "Unmute topic",
   "Unpin stream from top": "Unpin stream from top",
   "Unread": "Unread",
+  "Unread count summary (appears in desktop sidebar and browser tab)": "Unread count summary (appears in desktop sidebar and browser tab)",
   "Unstar": "Unstar",
   "Unstar all messages": "Unstar all messages",
   "Unstar all messages in topic": "Unstar all messages in topic",
   "Unstar messages in topic": "Unstar messages in topic",
   "Unsubscribe": "Unsubscribe",
-  "Unsubscribe from __stream_name__": "Unsubscribe from __stream_name__",
+  "Unsubscribe from {stream_name}": "Unsubscribe from {stream_name}",
   "Unsubscribed successfully!": "Unsubscribed successfully!",
   "Up to N minutes after posting": "Up to N minutes after posting",
-  "Up to __time_limit__ after posting": "Up to __time_limit__ after posting",
+  "Up to {time_limit} after posting": "Up to {time_limit} after posting",
+  "Update successful: Subdomains allowed for {domain}": "Update successful: Subdomains allowed for {domain}",
+  "Update successful: Subdomains no longer allowed for {domain}": "Update successful: Subdomains no longer allowed for {domain}",
   "Upload icon": "Upload icon",
   "Upload image or GIF": "Upload image or GIF",
   "Upload logo": "Upload logo",
   "Upload new profile picture": "Upload new profile picture",
   "Upload profile picture": "Upload profile picture",
-  "Upload stalled for %{seconds} seconds, aborting.": "Upload stalled for %{seconds} seconds, aborting.",
-  "Uploading __filename__\u2026": "Uploading __filename__\u2026",
+  "Upload stalled for %'{seconds}' seconds, aborting.": "Upload stalled for %'{seconds}' seconds, aborting.",
+  "Uploading {filename}\u2026": "Uploading {filename}\u2026",
   "Uploading\u2026": "Uploading\u2026",
   "Use full width on wide screens": "Use full width on wide screens",
   "Use organization level settings": "Use organization level settings",
   "User group added!": "User group added!",
-  "User groups allow you to <a href=\"/help/mention-a-user-or-group\" target=\"_blank\" rel=\"noopener noreferrer\">mention</a> multiple users at once. When you mention a user group, everyone in the group is notified as if they were individually mentioned.": "User groups allow you to <a href=\"/help/mention-a-user-or-group\" target=\"_blank\" rel=\"noopener noreferrer\">mention</a> multiple users at once. When you mention a user group, everyone in the group is notified as if they were individually mentioned.",
+  "User groups allow you to <z-link>mention</z-link> multiple users at once. When you mention a user group, everyone in the group is notified as if they were individually mentioned.": "User groups allow you to <z-link>mention</z-link> multiple users at once. When you mention a user group, everyone in the group is notified as if they were individually mentioned.",
   "User identity": "User identity",
   "User is already not subscribed.": "User is already not subscribed.",
   "User is deactivated": "User is deactivated",
@@ -716,8 +750,9 @@
   "View edit history": "View edit history",
   "View file": "View file",
   "View full profile": "View full profile",
-  "View in __name__": "View in __name__",
   "View in playground": "View in playground",
+  "View in {name}": "View in {name}",
+  "View in {playground_name}": "View in {playground_name}",
   "View messages sent": "View messages sent",
   "View private messages": "View private messages",
   "View private messages to myself": "View private messages to myself",
@@ -730,10 +765,11 @@
   "Visual desktop notifications": "Visual desktop notifications",
   "Waiting period (days)": "Waiting period (days)",
   "Waiting period before new members turn into full members": "Waiting period before new members turn into full members",
-  "Warning: <strong>__stream_name__</strong> is a private stream.": "Warning: <strong>__stream_name__</strong> is a private stream.",
+  "Warning: <strong>{stream_name}</strong> is a private stream.": "Warning: <strong>{stream_name}</strong> is a private stream.",
   "We are about to have a poll.  Please wait for the question.": "We are about to have a poll.  Please wait for the question.",
+  "We've replaced the \"{originalHotkey}\" hotkey with \"{replacementHotkey}\" to make this common shortcut easier to trigger.": "We've replaced the \"{originalHotkey}\" hotkey with \"{replacementHotkey}\" to make this common shortcut easier to trigger.",
   "Wednesday": "Wednesday",
-  "Week of __date__": "Week of __date__",
+  "Week of {date}": "Week of {date}",
   "Whether wildcard mentions like @all are treated as mentions for the purpose of notifications.": "Whether wildcard mentions like @all are treated as mentions for the purpose of notifications.",
   "Who can access the stream?": "Who can access the stream?",
   "Who can access user email addresses": "Who can access user email addresses",
@@ -750,32 +786,39 @@
   "Would you like to unstar all starred messages?  This action cannot be undone.": "Would you like to unstar all starred messages?  This action cannot be undone.",
   "Write": "Write",
   "Yes, send": "Yes, send",
-  "Yes, subscribe __count__ users!": "Yes, subscribe __count__ users!",
+  "Yes, subscribe {count} users!": "Yes, subscribe {count} users!",
   "Yes, unsubscribe": "Yes, unsubscribe",
   "Yes, unsubscribe from this stream": "Yes, unsubscribe from this stream",
   "Yes. Members and admins can send invitations.": "Yes. Members and admins can send invitations.",
   "Yes. Only admins can send invitations.": "Yes. Only admins can send invitations.",
   "Yesterday": "Yesterday",
-  "You (click to remove) reacted with __- emoji_name__": "You (click to remove) reacted with __- emoji_name__",
-  "You and __display_reply_to__": "You and __display_reply_to__",
-  "You and __recipients__": "You and __recipients__",
+  "You (click to remove) and {other_username} reacted with {emoji_name}": "You (click to remove) and {other_username} reacted with {emoji_name}",
+  "You (click to remove) reacted with {emoji_name}": "You (click to remove) reacted with {emoji_name}",
+  "You (click to remove), {comma_separated_usernames} and {last_username} reacted with {emoji_name}": "You (click to remove), {comma_separated_usernames} and {last_username} reacted with {emoji_name}",
+  "You and {display_reply_to}": "You and {display_reply_to}",
+  "You and {recipients}": "You and {recipients}",
   "You are not currently subscribed to this stream.": "You are not currently subscribed to this stream.",
-  "You are not subscribed to stream __stream__": "You are not subscribed to stream __stream__",
-  "You can reactivate deactivated users from <a href=\"#organization/deactivated-users-admin\">organization settings.</a>": "You can reactivate deactivated users from <a href=\"#organization/deactivated-users-admin\">organization settings.</a>",
+  "You are not subscribed to stream {stream}": "You are not subscribed to stream {stream}",
+  "You are searching for messages that are sent by more than one person, which is not possible.": "You are searching for messages that are sent by more than one person, which is not possible.",
+  "You are searching for messages that belong to more than one stream, which is not possible.": "You are searching for messages that belong to more than one stream, which is not possible.",
+  "You are searching for messages that belong to more than one topic, which is not possible.": "You are searching for messages that belong to more than one topic, which is not possible.",
+  "You can also make <z-link>tables</z-link> with this <z-link>Markdown-ish table syntax</z-link>.": "You can also make <z-link>tables</z-link> with this <z-link>Markdown-ish table syntax</z-link>.",
+  "You can reactivate deactivated users from <z-link>organization settings</z-link>.": "You can reactivate deactivated users from <z-link>organization settings</z-link>.",
   "You cannot create a stream with no subscribers!": "You cannot create a stream with no subscribers!",
   "You do not have permission to use wildcard mentions in this stream.": "You do not have permission to use wildcard mentions in this stream.",
   "You get": "You get",
-  "You have muted <strong><span class=\"stream\"></span> > <span class=\"topic\"></span></strong>.": "You have muted <strong><span class=\"stream\"></span> > <span class=\"topic\"></span></strong>.",
+  "You have muted <z-stream-topic></z-stream-topic>.": "You have muted <z-stream-topic></z-stream-topic>.",
   "You have no active bots.": "You have no active bots.",
   "You have no inactive bots.": "You have no inactive bots.",
   "You have not muted any topics yet.": "You have not muted any topics yet.",
   "You have not uploaded any files.": "You have not uploaded any files.",
   "You have nothing to send!": "You have nothing to send!",
+  "You must be an organization administrator to create a stream without subscribing.": "You must be an organization administrator to create a stream without subscribing.",
   "You need to be running Zephyr mirroring in order to send messages!": "You need to be running Zephyr mirroring in order to send messages!",
   "You searched for:": "You searched for:",
-  "You subscribed to stream __stream__": "You subscribed to stream __stream__",
+  "You subscribed to stream {stream}": "You subscribed to stream {stream}",
   "You type": "You type",
-  "You unsubscribed from stream __stream__": "You unsubscribed from stream __stream__",
+  "You unsubscribed from stream {stream}": "You unsubscribed from stream {stream}",
   "You're not subscribed to this stream. You will not be notified if other users reply to your message.": "You're not subscribed to this stream. You will not be notified if other users reply to your message.",
   "Your API key:": "Your API key:",
   "Your reminder note is empty!": "Your reminder note is empty!",
@@ -785,25 +828,31 @@
   "[More...]": "[More...]",
   "[Remove owner]": "[Remove owner]",
   "[Unset]": "[Unset]",
-  "__- username__ reacted with __- emoji_name__": "__- username__ reacted with __- emoji_name__",
-  "__days_old__ days ago": "__days_old__ days ago",
-  "__full_name__ is typing\u2026": "__full_name__ is typing\u2026",
-  "__hours__ hours ago": "__hours__ hours ago",
-  "__minutes__ min to edit": "__minutes__ min to edit",
-  "__minutes__ minutes ago": "__minutes__ minutes ago",
-  "__seconds__ sec to edit": "__seconds__ sec to edit",
-  "__starred_status__ this message": "__starred_status__ this message",
-  "__starred_status__ this message (Ctrl + s)": "__starred_status__ this message (Ctrl + s)",
   "and": "and",
   "beta": "beta",
   "clear": "clear",
   "cookie": "cookie",
+  "group private messages with {recipient}": "group private messages with {recipient}",
   "in 1 hour": "in 1 hour",
   "in 20 minutes": "in 20 minutes",
   "in 3 hours": "in 3 hours",
   "leafy green vegetable": "leafy green vegetable",
   "marketing": "marketing",
   "more topics": "more topics",
-  "private messages with __- recipient__": "private messages with __- recipient__",
-  "private messages with yourself": "private messages with yourself"
+  "private messages with yourself": "private messages with yourself",
+  "private messages with {recipient}": "private messages with {recipient}",
+  "{comma_separated_usernames} and {last_username} reacted with {emoji_name}": "{comma_separated_usernames} and {last_username} reacted with {emoji_name}",
+  "{count} users are subscribed to #{title}": "{count} users are subscribed to #{title}",
+  "{days_old} days ago": "{days_old} days ago",
+  "{full_name} is typing\u2026": "{full_name} is typing\u2026",
+  "{hours} hours ago": "{hours} hours ago",
+  "{last_active_date}": "{last_active_date}",
+  "{minutes} min to edit": "{minutes} min to edit",
+  "{minutes} minutes ago": "{minutes} minutes ago",
+  "{seconds} sec to edit": "{seconds} sec to edit",
+  "{starred_status} this message": "{starred_status} this message",
+  "{starred_status} this message (Ctrl + s)": "{starred_status} this message (Ctrl + s)",
+  "{username} [said]({link_to_message}):": "{username} [said]({link_to_message}):",
+  "{username} reacted with {emoji_name}": "{username} reacted with {emoji_name}",
+  "{wildcard_mention_token} (Notify stream)": "{wildcard_mention_token} (Notify stream)"
 }
\ No newline at end of file
```

</details>

Reply:
e710110a9ea94716b4eeca396e6f8fb4c62c02d3 is only half of the background; the other is 134b165b1a2d8bdf641909c0e8306cc5226d6eea. Regardless of background, my claim is that the `\n` → `\\n` → `\n` replacements as currently written cancel in a way that has nothing to do with i18next or FormatJS.